src/net.h

```diff
@@ -421,8 +500,10 @@ class CNode
 
     const ConnectionType m_conn_type;
 
-    /** Move all messages from the received queue to the processing queue. */
-    void MarkReceivedMsgsForProcessing()
+    /** Move all messages from the received queue to the processing queue.
+     * Also update the global map of network message statistics.
+    */
+    void MarkReceivedMsgsForProcessing(NetStats& net_stats)
```

Comment:
Should I just pass a reference of `NetStats` to every `CNode` upon initialization? 

Example: https://github.com/bitcoin/bitcoin/pull/27324/commits/cd0c8eeb0940790b6ba83786d1c9e362d4dc4829